### PR TITLE
configchange: proceed with deployment with non-automatic ICTs

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/openshift/origin/issues/7663
Fixes https://github.com/openshift/origin/issues/7686

@ironcladlou @smarterclayton

### CODE REVIEW

**Comment by kargakis** [comment]
@mfojtik can you test this and verify that https://github.com/openshift/origin/issues/7686 is fixed?

**Comment by mfojtik** [comment]
@kargakis will do

**Comment by mfojtik** [comment]
magic! https://github.com/openshift/origin/issues/7686 is fixed by this.

**Comment by ironcladlou** [comment]
LGTM, better than the idea I proposed in #7663

**Comment by kargakis** [comment]
[merge]

File: pkg/deploy/util/util.go [hunk 1]
```diff
@@ -104,6 +104,27 @@ func HasChangeTrigger(config *deployapi.DeploymentConfig) bool {
...
+	for _, trigger := range config.Spec.Triggers {
+		if trigger.Type == deployapi.DeploymentTriggerOnImageChange {
+			return true
+		}
+	}
+	return false
+}
+
+// GetImageChangeTrigger returns the image change trigger from a deployment config.
+func GetImageChangeTrigger(config *deployapi.DeploymentConfig) deployapi.DeploymentTriggerPolicy {
```
Comment (smarterclayton) [inline pkg/deploy/util/util.go]: A deployment config has multiple triggers, so this is not correct if only the second or third trigger is automatic. Please add a test and update this.

**Pull request merged**